### PR TITLE
Raise SSHCommandError from a shorter version of the ssh() function

### PR DESCRIPTION
**Pull request by stormi**
pytest always prints the whole ssh() function body because that's where
the exceptions are raised. Trick it into displaying a shorter version of
the ssh() function by moving all code into a _ssh() function and only
handling returning results and raising Exceptions in ssh().

Signed-off-by: Samuel Verschelde <stormi-xcp@ylix.fr>

### CODE REVIEW

**Comment by stormi** [comment]
Here's an example of what a command failure looks like now:
```
========================================================== FAILURES ===========================================================
________________________________________ TestBasicNoSSH.test_start[172.16.210.22-None] ________________________________________

self = <test_basic_without_ssh.TestBasicNoSSH object at 0x7f48fd7ed940>, imported_vm = <lib.vm.VM object at 0x7f48fd7ed640>

    def test_start(self, imported_vm):
        vm = imported_vm
        # if VM already running, stop it
        if (vm.is_running()):
            logging.info("VM already running, shutting it down first")
            vm.shutdown(verify=True)
        vm.start()
>       vm.host.ssh(["ça peut pas marcher ça"])

tests/misc/test_basic_without_ssh.py:37:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
lib/host.py:33: in ssh
    return commands.ssh(self.hostname_or_ip, cmd, check=check, simple_output=simple_output,
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

hostname_or_ip = '172.16.210.22', cmd = ['ça peut pas marcher ça'], check = True, simple_output = True
suppress_fingerprint_warnings = True, background = False, target_os = 'linux', decode = True

    def ssh(hostname_or_ip, cmd, check=True, simple_output=True, suppress_fingerprint_warnings=True,
            background=False, target_os='linux', decode=True):
        success, result_or_exc = _ssh(hostname_or_ip, cmd, check, simple_output, suppress_fingerprint_warnings,
                                      background, target_os, decode)
        if success:
            return result_or_exc
        else:
>           raise result_or_exc
E           lib.commands.SSHCommandFailed: SSH command (ça peut pas marcher ça) failed with return code 127: bash: ça: command not found

lib/commands.py:147: SSHCommandFailed
```

File: lib/commands.py [hunk 1]
```diff
@@ -115,23 +116,36 @@ def ssh(hostname_or_ip, cmd, check=True, simple_output=True, suppress_fingerprin
...
+# This function is kept short for shorter pytest traces upon SSH failures, which are common,
+# as pytest prints the whole function definition that raised the SSHCommandFailed exception
+def ssh(hostname_or_ip, cmd, check=True, simple_output=True, suppress_fingerprint_warnings=True,
+        background=False, target_os='linux', decode=True):
+    success, result_or_exc = _ssh(hostname_or_ip, cmd, check, simple_output, suppress_fingerprint_warnings,
+                                  background, target_os, decode)
+    if success:
+        return result_or_exc
+    else:
+        raise result_or_exc
```
[the remark this answers is not in the document]
Comment (benjamreis) [reply lib/commands.py:147]: Also you might want to use `type(result_or_exc)` instead of returning a `Boolean`

File: lib/commands.py [hunk 2]
```diff
@@ -115,23 +116,36 @@ def ssh(hostname_or_ip, cmd, check=True, simple_output=True, suppress_fingerprin
...
+
```
Comment (benjamreis) [inline lib/commands.py:148]: Useless extraline.

**Review by benjamreis — CHANGES_REQUESTED** [review]

File: lib/commands.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (stormi) [reply lib/commands.py]: Done, sort of.

**Review by benjamreis — APPROVED** [review]

**Review by Wescoeur — APPROVED** [review]

**Pull request merged**